### PR TITLE
🐛 datcore-adapter (Pennsieve API): filter out deleted files

### PR DESCRIPTION
**Pull request by elisabettai**
## What do these changes do?
 
When querying dataset packages, e.g. with:
```
GET https://api.pennsieve.io/datasets/N:dataset:a20b0a68-11be-4c39-a65a-15903de5edc2/packages
```
There might be deleted files in the response, e.g.:
```
{
      "content": {
        "id": 3706023,
        "name": "__DELETED__N:package:268f6590-c5d3-4efd-9538-8e6c3da77f54_submission.xlsx",
        "packageType": "Unsupported",
        "datasetId": 2998,
        "datasetNodeId": "N:dataset:a20b0a68-11be-4c39-a65a-15903de5edc2",
        "ownerId": 832,
        "state": "DELETED",
        "createdAt": "2023-12-28T19:28:06.491464Z",
        "updatedAt": "2023-12-28T19:32:38.143404Z",
        "nodeId": "N:package:268f6590-c5d3-4efd-9538-8e6c3da77f54"
      },
      "properties": [
        {
          "category": "Pennsieve",
          "properties": [
            {
              "key": "subtype",
              "value": "MS Excel",
              "dataType": "string",
              "fixed": false,
              "hidden": true,
              "display": "MS Excel"
            },
            {
              "key": "icon",
              "value": "Excel",
              "dataType": "string",
              "fixed": false,
              "hidden": true,
              "display": "Excel"
            }
          ]
        }
      ],
      "children": [],
      "extension": "xlsx"
    },
```
Once we then try to get those files, a 404 error is raised. 
At the moment, the Pennsieve API does not have a flag to exclude those files, so we need to filter them out on our end (this was confirmed by Pennsieve devs).

## Related issue/s

Fixes https://github.com/ITISFoundation/osparc-issues/issues/1231
(Original support request came via the oSPARC Support Google form, see details here https://z43.fogbugz.com/f/cases/192158)

## How to test

**Manually**
1. Use the oSPARC UI to navigate the datcore data tree.
2. Before this fix: the dataset "Influence of vagus nerve stimulation on cardiac activity in pigs" will load forever
3. After the fix: the dataset "Influence of vagus nerve stimulation on cardiac activity in pigs" works

**Unit tests**
Run tests with your API credentials and `--dataset=dddbce01-bae0-4e2b-85fd-dafb47182afa` 

## Dev Checklist

- [x] No ENV changes or I properly updated ENV ([read the instruction](https://git.speag.com/oSparc/osparc-ops-deployment-configuration/-/blob/configs/README.md?ref_type=heads#how-to-update-env-variables))

## DevOps Checklist
Not applicable (at least for this change). 😉

### CODE REVIEW

**Review by pcrespov — APPROVED** [review]
thx!

File: services/datcore-adapter/src/simcore_service_datcore_adapter/modules/pennsieve.py [hunk 1]
```diff
@@ -181,7 +181,7 @@ async def _get_dataset_packages(
         page_size: int,
         cursor: str,
     ) -> dict[str, Any]:
-        return cast(
+        package = cast(
```
Comment (GitHK) [inline services/datcore-adapter/src/simcore_service_datcore_adapter/modules/pennsieve.py:184]: Mintor: I think plural would sound better here

**Review by GitHK — APPROVED** [review]

File: services/datcore-adapter/src/simcore_service_datcore_adapter/modules/pennsieve.py (hunk 1 above)
Comment (elisabettai) [reply services/datcore-adapter/src/simcore_service_datcore_adapter/modules/pennsieve.py]: True, thanks!

**Review by sanderegg — APPROVED** [review]
perfect! thanks!!

**Pull request merged**